messaginginapp/src/main/java/io/customer/messaginginapp/gist/data/listeners/Queue.kt

```diff
@@ -22,18 +22,25 @@ import okhttp3.ResponseBody.Companion.toResponseBody
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 
-class Queue : GistListener {
+interface GistQueue {
+    fun fetchUserMessages()
+    fun clearPrefs(context: Context)
+    fun logView(message: Message)
+}
 
-    internal var localMessageStore: MutableList<Message> = mutableListOf()
-    internal var shownMessageQueueIds = mutableSetOf<String>()
-    private val inAppMessagingManager: InAppMessagingManager = SDKComponent.inAppMessagingManager
+class Queue : GistQueue {
 
-    init {
-        GistSdk.addListener(this)
-    }
+    private val inAppMessagingManager: InAppMessagingManager
+        get() = SDKComponent.inAppMessagingManager
+    private val state: InAppMessagingState
+        get() = inAppMessagingManager.getCurrentState()
+    private val logger: Logger = SDKComponent.logger
+    private val scope: CoroutineScope = SDKComponent.scopeProvider.lifecycleListenerScope
```

Comment:
I'm assuming this will be replaced with dedicated scope in future PRs

Reply:
yup